src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -401,7 +430,54 @@ def get_run(self, run_id: int) -> Optional[Run]:
             if run_id not in self.run_ids:
                 log(ERROR, "`run_id` is invalid")
                 return None
-            return self.run_ids[run_id]
+            return self.run_ids[run_id].run
+
+    def get_run_status(self, run_ids: set[int]) -> dict[int, RunStatus]:
```

Comment:
(very minor) Would it be better to assume `run_ids` is a list, and then construct a set out of it and proceed with the rest of the logic? The return dict will be the same but we'll prevent unnecessary iterations if `run_ids` contains duplicated ids for some reason.

Reply:
That makes sense.